nengo_bones/templates/.travis.yml.template

```diff
@@ -0,0 +1,131 @@
+# Automatically generated by nengo-bones, do not edit this file directly
+# Version: {{ version }}
+
+language: python
+python:
+  - "3.6"
+notifications:
+  email:
+    on_success: change
+    on_failure: change
+# cache:
+#   directories:
+#   - $HOME/miniconda
+
+dist: trusty
+
+env:
+  global:
+    - SCRIPT="test"
+    - PYTHON_VERSION="{{ python_version }}"
+    - PIP_UPGRADE="true"  # always upgrade to latest version
+    - PIP_UPGRADE_STRATEGY="eager"  # upgrade all dependencies
+    - TEST_ARGS=""
+    - COV_CORE_SOURCE={{ pkg_name }}  # early start pytest-cov engine
+    - COV_CORE_CONFIG=.coveragerc
+    - COV_CORE_DATAFILE=.coverage.eager
+    {% for var, val in global_vars.items() %}
+    - {{ var|upper }}="{{ val }}"
+    {% endfor %}
+
+jobs:
+  include:
+  {% for job in jobs %}
+  -
+    {% if job["env"] %}
+    env:
+      {% for var, val in job["env"].items() %}
+      {{ var|upper }}="{{ val }}"
+      {% endfor %}
+    {% endif %}
+    {% for key, val in job.items() if key != "env" %}
+    {{ key }}: {{ val }}
+    {% endfor %}
+  {% endfor %}
+  {% if pypi_user %}
+  - stage: deploy
+    if: branch =~ ^release-candidate-* OR tag =~ ^v[0-9]*
+    env: SCRIPT="deploy"
+    cache: false
+    deploy:
+      - provider: pypi
+        server: https://test.pypi.org/legacy/
+        user: {{ pypi_user }}
+        password: $PYPI_TOKEN
+        distributions: "{% for dist in deploy_dists -%} {{ dist }} {% endfor %}"
+        on:
+          all_branches: true
+          condition: $TRAVIS_BRANCH =~ ^release-candidate-*
+          condition: $TRAVIS_TAG = ""
+      - provider: pypi
+        user: {{ pypi_user }}
+        password: $PYPI_TOKEN
+        distributions: "{% for dist in deploy_dists -%} {{ dist }} {% endfor %}"
+        on:
+          all_branches: true
+          condition: $TRAVIS_TAG =~ ^v[0-9]*
+  {% endif %}
+
+before_install:
+  # export travis_terminate for use in scripts
+  - export -f travis_terminate
+  - export -f _travis_terminate_linux
+  - export -f _travis_terminate_osx
+  - export -f _travis_terminate_unix
+  - export -f _travis_terminate_windows
+  # install/run nengo-bones
+  - pip install git+https://github.com/nengo/nengo-bones.git@template-ci
+  - generate-bones --output-dir .ci --template-dir .ci ci-scripts
+  - check-bones
```

Comment:
The mistake I'd be trying to prevent here is doing dev-work on nengo-bones (where the version isn't changing) but then forgetting to apply bones to self. Since the unit tests don't look at the actual yml files, they could still pass. But fair enough!

Reply:
The other mistake this would safeguard is updating `.nengobones.yml` for some project but forgetting to run `generate-bones`.

Reply:
Yeah I'm convinced, I think that'd be good.  To summarize, the idea is we have `check-bones` generate all the manual files, hash them (I think that'll be easier than comparing diffs), compare them to the existing files, and error if they don't match.

I'd probably be inclined to make an issue to do that after merging this in though, so that we can get all the other PRs merged in before introducing more changes.  And this would only involve changes in `nengo-bones`, so it'll be easy to change afterwards (don't have to make any downstream PRs).

Reply:
It might also not be so fun right now for `nengo-loihi` since `check-bones` will be run on Python 3.5 and give non-deterministic `.yml` files...

Reply:
The `check-bones`/`generate-bones` scripts actually run outside the conda environment, using the base travis-ci python (which we set to 3.6).  For exactly that reason, so that `nengo-bones` isn't tied to the build environment of downstream repos.